pkg/api/handlers/mcpcatalogs.go

```diff
@@ -1164,19 +1161,10 @@ func (h *MCPCatalogHandler) GenerateComponentToolPreviews(req api.Context) error
 		return fmt.Errorf("failed to generate tool preview: %w", err)
 	}
 
-	// Return the tool previews on a skeleton entry
-	// We don't bother adding these to the real entry because:
-	// - it may no longer exist
-	// - we already have enough information to generate composite tool overrides for the component
-	entry := v1.MCPServerCatalogEntry{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      component.CatalogEntryID,
-			Namespace: composite.Namespace,
-		},
-		Spec: v1.MCPServerCatalogEntrySpec{
-			MCPCatalogName: composite.Spec.MCPCatalogName,
-			Manifest:       component.Manifest,
-		},
+	// Load the standalone catalog entry for shape / metadata and attach the preview.
+	var entry v1.MCPServerCatalogEntry
+	if err := req.Get(&entry, componentID); err != nil {
+		return fmt.Errorf("failed to get component catalog entry: %w", err)
```

Comment:
This is undoing a change I recently made that fixes https://github.com/obot-platform/obot/issues/4983.

We want to be able to configure tool overrides for components whose original catalog entry no longer exists.

In general, we should be able to make changes to the composite entry and continue using it (creating servers with it, updating them, etc) even if the original component entries do not exist anymore; at least until the entry is "upgraded" and the component with missing entries are removed.

That being said, this feature may be more trouble than it's worth (especially since there are a few other bugs open that relate to missing parent entries for component servers).